odxtools/nameditemlist.py

```diff
@@ -119,11 +113,11 @@ def __getitem__(self, key: str) -> T:
     def __getitem__(self, key: slice) -> List[T]:
         ...
 
-    def __getitem__(self, key: Union[int, str, slice]) -> Union[T, List[T]]:
-        if isinstance(key, int):
-            return self._item_list[key]
+    def __getitem__(self, key: Union[SupportsIndex, str, slice]) -> Union[T, List[T]]:
+        if isinstance(key, SupportsIndex):
+            return self.data[key]
         elif isinstance(key, slice):
-            return self._item_list[key]
+            return self.data[key]
         else:
             return self._item_dict[key]
```

Comment:
nitpick: function could be simpler by using the divt when the key is str, otherwise fallback to super

Reply:
I tried this. this leads to mypy complaints:
```
odxtools/nameditemlist.py:120: error: Incompatible return value type (got "Any | ItemAttributeList[T]", expected "T | list[T]")  [return-value]
odxtools/nameditemlist.py:155: error: Argument 1 of "_get_item_key" is incompatible with supertype "ItemAttributeList"; supertype defines the argument type as "T"  [override]
odxtools/nameditemlist.py:155: note: This violates the Liskov substitution principle
odxtools/nameditemlist.py:155: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
```

I think that these would be fixed if `ItemAttributeList` would directly derive from `UserList[T]`, but this is not allowed in python-3.8...